revolution_pi/gpio_pin.go

```diff
@@ -11,37 +11,83 @@ import (
 	"unsafe"
 )
 
+const (
+	outputPWMActiveOffset    = 110 // offset for reading a PWM frequency from a DIO
+	outputPWMFrequencyOffset = 112
+	outputWordToPWMOffset    = 2
+	inputWordToCounterOffset = 6
+	dioMemoryOffset          = 88
+)
+
 type gpioPin struct {
-	Name        string // Variable name
-	Address     uint16 // Address of the byte in the process image
-	BitPosition uint8  // 0-7 bit position, >= 8 whole byte
-	Length      uint16 // length of the variable in bits. Possible values are 1, 8, 16 and 32
-	ControlChip *gpioChip
-	pwmMode     bool
-	initialized bool
+	Name         string // Variable name
+	Address      uint16 // Address of the byte in the process image
+	BitPosition  uint8  // 0-7 bit position, >= 8 whole byte
+	Length       uint16 // length of the variable in bits. Possible values are 1, 8, 16 and 32
+	ControlChip  *gpioChip
+	pwmMode      bool
+	initialized  bool
+	outputOffset uint16
+	inputOffset  uint16
 }
 
 func (pin *gpioPin) initialize() error {
-	val, err := pin.ControlChip.getBitValue(int64(110+(pin.Address-70)), pin.BitPosition)
+	dio, err := pin.ControlChip.findDIODevice(pin.Address)
```

Comment:
this should probably live in `gpio_chip.GetGPIOPin`, but I will hold off until I go to setup AIO devices